sdks/typescript/src/v1/declaration.ts

```diff
@@ -300,11 +340,10 @@ export class BaseWorkflowDeclaration<
       throw UNBOUND_ERR;
     }
 
-    if (Array.isArray(input)) {
-      return Promise.all(input.map((i) => this.runAndWait(i, options, _standaloneTaskName)));
-    }
-
-    return this.run(input, options, _standaloneTaskName);
+    // note: typescript is not smart enough to infer that input is an array
+    return Array.isArray(input)
+      ? this.run(input, options, _standaloneTaskName)
+      : this.run(input, options, _standaloneTaskName);
```

Comment:
we do the same thing in either case? if yes, can we simplify this to `this.run(...)` instead without the ternary?

Reply:
typescript is exploding without this and i prefer this over casting

Reply:
😭